src/app/core/admin/admin-entity/admin-entity-general-settings/admin-entity-general-settings.component.ts

```diff
@@ -77,29 +73,40 @@ export class AdminEntityGeneralSettingsComponent implements OnChanges, OnInit {
     this.form = this.fb.group({
       basicSettings: this.basicSettingsForm,
     });
+    if (this.usedFields) {
+      this.initAvailableDatatypes(this.usedFields);
+    }
 
     this.form.valueChanges.subscribe((value) => {
       this.emitStaticDetails(); // Optionally, emit the initial value
     });
   }
+  private initAvailableDatatypes(array) {
+    const allUsedFields: string[] = [];
 
-  emitStaticDetails() {
-    const toStringAttributesControl =
-      this.basicSettingsForm.get("toStringAttributes");
-    let toStringAttributesValue = toStringAttributesControl.value;
-    // Convert toStringAttributesValue to an array if it's a string
-    if (typeof toStringAttributesValue === "string") {
-      toStringAttributesValue = toStringAttributesValue
-        .split(",")
-        .map((item) => item.trim());
-    }
-
-    // Update the form control with the modified value
-    toStringAttributesControl.setValue(toStringAttributesValue, {
-      emitEvent: false,
+    const basicInformationPanel = array.panels.find(
+      (panel) => panel.title === "Basic Information",
+    );
+    basicInformationPanel.components.forEach((component) => {
+      component.config.fieldGroups.forEach((fieldGroup) => {
+        allUsedFields.push(...fieldGroup.fields);
+      });
     });
```

Comment:
This won't work, because panel titles can be renamed to anything. Also, while it may be useful to only offer used fields, I would rather display all fields of this entity type here.
Use entityConstructor.schema to get the fields here (and hide those without label, because these are internal, technical fields): `allFields = [...this.entityConstructor.schema.values()].filter((field) => field.label)`

Reply:
updated!